packages/web-app-files/src/helpers/resources.js

```diff
@@ -168,20 +168,50 @@ export function buildSpace(space) {
     spaceRoles,
     spaceImageData,
     spaceReadmeData,
-    canUpload: function () {
-      return true
+    canUpload: function ({ user }) {
+      const allowedRoles = [
+        ...this.spaceRoles[spaceRoleManager.name],
+        ...this.spaceRoles[spaceRoleEditor.name]
+      ]
+      return user && allowedRoles.includes(user.uuid)
     },
     canDownload: function () {
       return true
     },
-    canBeDeleted: function () {
-      return true
+    canBeDeleted: function ({ user }) {
+      return this.disabled && user && this.spaceRoles[spaceRoleManager.name].includes(user.uuid)
     },
-    canRename: function () {
-      return true
+    canRename: function ({ user }) {
+      return user && this.spaceRoles[spaceRoleManager.name].includes(user.uuid)
```

Comment:
Are there any restrictions if the space is disabled ?

Reply:
Currently not, all actions are still allowed for disabled spaces